packages/git/src/browser/git-contribution.ts

```diff
@@ -440,6 +441,19 @@ export class GitContribution implements CommandContribution, MenuContribution, T
         registry.registerCommand(GIT_COMMANDS.OPEN_CHANGED_FILE, {
             execute: (...arg: ScmResource[]) => {
                 for (const resource of arg) {
+                    const provider = this.repositoryProvider.selectedScmProvider;
+                    if (provider) {
+                        const gitStatus = provider.getStatus();
+                        if (gitStatus) {
+                            const deletedFile = gitStatus.changes.find(change =>
+                                change.status === GitFileStatus.Deleted && change.uri === resource.sourceUri.toString());
```

Comment:
I wonder whether we should fix `resource.sourceUri` in first place instead? cc @westbury

Reply:
So, what's the conclusion?

Reply:
please see https://github.com/eclipse-theia/theia/pull/8107#discussion_r448153652